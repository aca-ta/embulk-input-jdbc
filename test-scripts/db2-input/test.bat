CALL db2setcp
CALL db2 "CONNECT TO TESTDB USER db2admin USING XXXXXXXX"
CALL db2 "DROP TABLE EMBULK_INPUT"
CALL db2 "CREATE TABLE EMBULK_INPUT (ID INT NOT NULL, NUM DECIMAL(12,2), STR CHAR(8), VARSTR VARCHAR(8), DT DATE, TS TIMESTAMP, TS9 TIMESTAMP(9), PRIMARY KEY(ID))"
CALL db2 "INSERT INTO EMBULK_INPUT VALUES(1, 123.4, 'test1', 'TEST1', '2016-10-13', '2016-10-14 01:02:03', '2016-10-15 01:02:03.123456789')"
CALL db2 "INSERT INTO EMBULK_INPUT VALUES(2, 1234567890.12, 'test9999', 'TEST9999', '2016-12-31', '2016-12-31 23:59:59', '2016-12-31 23:59:59.999999999')"
CALL db2 "INSERT INTO EMBULK_INPUT VALUES(3, NULL, NULL, NULL, NULL, NULL, NULL)"

del data\test000.00.csv
CALL embulk run test.yml

echo "diff data/test_expected.csv data/test000.00.csv"
diff data/test_expected.csv data/test000.00.csv

IF "%ERRORLEVEL%" == "0" (ECHO "OK!") ELSE (ECHO "embulk-input-db2 FAILED!")
