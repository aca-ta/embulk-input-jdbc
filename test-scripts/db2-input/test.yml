in:
  type: db2
  driver_path: driver/db2jcc4.jar
  host: localhost
  database: TESTDB
  user: db2admin
  password: XXXXXXXX
  table: EMBULK_INPUT
  select: "*"
  column_options:
    ID: {type: long}
    NUM: {type: string}
    STR: {type: string}
    VARSTR: {type: string}
    DT: {type: string, timestamp_format: '%Y-%m-%d', timezone: "+0900"}
    TS: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S', timezone: "+0900"}
    TS9: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S.%9N', timezone: "+0900"}
out:
  type: file
  path_prefix: data/test
  file_ext: csv
  formatter:
    type: csv
