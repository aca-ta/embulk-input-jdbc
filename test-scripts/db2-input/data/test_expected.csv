ID,NUM,STR,VARSTR,DT,TS,TS9
1,123.40,test1   ,TEST1,2016-10-13,2016-10-14 01:02:03,2016-10-15 01:02:03.123456789
2,1234567890.12,test9999,TEST9999,2016-12-31,2016-12-31 23:59:59,2016-12-31 23:59:59.999999999
3,,,,,,
